app/controllers/queue_items_controller.rb

```diff
@@ -0,0 +1,45 @@
+class QueueItemsController < ApplicationController
+  before_action :require_login
+
+  def index
+    @items = current_user.queue_items
+    unless logged_in?
+      redirect_to log_in_path
+    end
+  end
+
+  def create
+    video = Video.find(params[:video_id])
+    queue_video(video) unless not_in_queue?(video)
+    redirect_to my_queue_path
+  end
+
+  def destroy
+    item = QueueItem.find(params[:id])
+    if current_user.queue_items.include? item
+      item.destroy
+      update_list_orders
+    end
+    redirect_to my_queue_path
+  end
+
+  private
+
+  def update_list_orders
+    current_user.queue_items.each_with_index do |item, index|
+      item.update_attribute(:list_order, index + 1)
+    end
+  end
+
+  def queue_video(video)
+    QueueItem.create(list_order: get_list_order, video: video, user: current_user)
+  end
+
+  def get_list_order
+    current_user.queue_items.size + 1
+  end
+
+  def not_in_queue?(video)
```

Comment:
This methods name is confusing - it points to exact opposite. Because `not_in_queue?` means it is not in this queue? so it should be able to be queued right. If it returns `true` - that it is NOT queued it will not be used right in your logic with `unless` as `unless true` will not follow to `queue_video`. I recommend to call this method `has_already_queued?(video)` and stick it to User's model as it can be reused in views too.